# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/TextUpdater.cpp
# Commit: e488a2686d0b
# Full Hash: e488a2686d0b6354c723dd896c6ce20749c8b3d7
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-12-06 03:46:09
# Regressor Bug: 1766794
# File Overlap Count: 1
# Description:
#   Bug 1802268 part 2: Reinstate partial invalidation of cached HyperText offsets for local HyperTextAccessible. r=eeejay
#   
#   In bug 1766794, I replaced partial invalidation of cached offsets with full cache invalidation.
#   I did this to simplify the code, since the offsets cache was being extended to RemoteAccessible.
#   This is fine for RemoteAccessible because offsets are only needed there when clients query them.
# ==============================================================================

diff -r 80097f40fc50 -r e488a2686d0b accessible/base/TextUpdater.cpp
--- a/accessible/base/TextUpdater.cpp	Mon Dec 05 23:27:23 2022 +0000
+++ b/accessible/base/TextUpdater.cpp	Mon Dec 05 23:27:24 2022 +0000
@@ -47,7 +47,8 @@
     return;
   }
 
-  mTextOffset = mHyperText->GetChildOffset(mTextLeaf);
+  // Get the text leaf accessible offset and invalidate cached offsets after it.
+  mTextOffset = mHyperText->GetChildOffset(mTextLeaf, true);
   NS_ASSERTION(mTextOffset != -1, "Text leaf hasn't offset within hyper text!");
 
   // Don't bother diffing if the hypertext isn't editable. Diffing non-editable
@@ -71,7 +72,6 @@
 
     // Update the text.
     mTextLeaf->SetText(aNewText);
-    mHyperText->InvalidateCachedHyperTextOffsets();
     return;
   }
 
@@ -117,7 +117,6 @@
 
     // Update the text.
     mTextLeaf->SetText(aNewText);
-    mHyperText->InvalidateCachedHyperTextOffsets();
     return;
   }
 
@@ -163,7 +162,6 @@
 
   // Update the text.
   mTextLeaf->SetText(aNewText);
-  mHyperText->InvalidateCachedHyperTextOffsets();
 }
 
 void TextUpdater::ComputeTextChangeEvents(